.github/workflows/root-ci.yml

```diff
@@ -75,6 +75,10 @@ env:
   OS_INTERFACE: 'public'
   OS_REGION_NAME: 'cern'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
```

Comment:
As is this setting will apply to all the workflow (see similar discussion at https://github.com/celeritas-project/celeritas/pull/1020/files#r1399421620).  Is that what we want?

Reply:
I am not sure what was discussed there and in which context but this change stops the current ongoing build of a PR upon push and starts a new one. I think that’s what we want most of the time and that’s consistent to our jenkins infrastructure. 

We might want to limit that to PRs only.

Reply:
It also applies to branches where we do *not* want to cancel builds. There was a lengthy discussion on Mattermost with the preliminary outcome that it's hard to implement cancellation only for PRs...

Reply:
See my suggestion.

Reply:
I just noticed that GitHub updated their documentation and provides a recipe to implement what we need: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-fallback-value

So what we want is probably:
```
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.run_id }}
  cancel-in-progress: true
```

The alternative with `cancel-in-progress: ${{ github.event_name == 'pull_request' }}` probably also works, but I don't know what are the consequences of using the undefined `github.event.pull_request.number` on branches and putting all workflow runs into a single group...

Reply:
If you are sure that what you suggest is better - I can put it in.

Reply:
So I just checked both options in https://github.com/hahnjo/github-actions-test: They work equally well for PRs, but the consequence of putting all non-PR workflows into a single concurrency group means that only one of them is running at a time. For example, https://github.com/hahnjo/github-actions-test/actions/runs/7722867934 was pending while the previous https://github.com/hahnjo/github-actions-test/actions/runs/7722862441 was still running. That's a problem because it mixes together all branches and other ways of triggering a workflow (scheduled nightly, release), and even worse because of our heterogeneous runners and we want fast Linux runners to start building the next commit while Windows and macOS is still working on (multiple) others.

tl;dr I think `group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.run_id }}` gives us what we want, at least in the scenarios I tested.